### DISCUSSION TITLE
David Whittaker SPC Format

### DISCUSSION THREAD

**Discussion by guylamar2006** [vgmtrans/vgmtrans#554] [Music formats]
This would be a neat format to support transcribing music from David whittaker games like lemmings, Krusty's Super Fun House, battle cars, dream TV, gods, Batman revenge of the joker, etc.

Here's info on the sound engine: 

https://sneslab.net/wiki/David_Whittaker_Sound_Engine

**Comment by KungFuFurby** [comment]
I see you're working on implementing the format itself, going by the fork that I found. I'm here if you need some clarification on the format itself (or just differences in general), given that I actually wrote the article in question for the most part. To put things simply, you have *a lot* of builds to sort out as well, and I know of a few that don't even support music.

I think the variable ARAM map should be mostly consistent for the middle set of versions, though I know it shifted at least once. I haven't traced them all the way through, but the earliest ones (referring to Dream TV Beta 2, Kick Off and Krusty's Super Fun House) have more drastic differences due to supporting less tracks overall, and I think the latest ones (referring to Shaq Fu and Michael Jordan: Chaos in the Windy City) also have variable ARAM map differences, just in case you're trying to use those locations for assisting with song data extraction for conversion.

**Comment by guylamar2006** [comment]
Been spending the last few days trying to figure out where the vcmds start at so I can try to figure out what to add to the WhitakerSnesScanner, starting with my favorite game Krusty's Super Fun House, I'm refactoring NinSnes as a base since the instrument format is N-SPC/Kankichi-kun compatible minus noise support for SRCN above 127. Starting out by renaming NinSnes to WhittakerSnes in the whittaker-nspc branch, I'm also getting into SPC700 disassembly but I'm trying to figure out how to convert disassembled db and dw back to DB dw due to a bug in spcdas. 

________________________________

**Comment by KungFuFurby** [comment]
There are several differences. The instrument format is the closest match, as well as the pitch code... but everything else I think you're going to find out is not as clean-cut as you'd expect it to be.

I'm going to direct you to some additional resources, because I've done some disassemblies here (as well as extracted the entire collection of binaries): https://github.com/KungFuFurby/vgm-disasm/tree/david-whittaker-disasm/snes/David%20Whittaker

The disassemblies aren't actually commented, though, but I did the earlier builds, including Krusty's Super Fun House. I tend to do disassemblies strategically since I can otherwise read the raw opcodes in a hex editor for the most part, and a hex editor comparison between two files can usually grab most code differences between the two.

I also have a partial ARAM map trace here, mainly covering the earlier builds (and again covering Krusty's Super Fun House): https://github.com/KungFuFurby/SNESSoundDriverDocViewer/tree/raw-notes/Raw%20Notes/David%20Whittaker
